src/librustdoc/clean/mod.rs

```diff
@@ -1937,6 +1937,7 @@ impl Clean<Span> for rustc_span::Span {
         let hi = sm.lookup_char_pos(self.hi());
         Span {
             filename,
+            cnum: lo.file.cnum,
```

Comment:
Heh this is really simple. Although, perhaps a better thing to do would be to keep `lo.file`? And remove the `span_to_filename` call, which is then wasteful.

Also cc @Aaron1011 I hope this will continue to work.

Reply:
That's actually what I originally tried but it ended up quite a bit more complicated than this. The main advantage to keeping the full `SourceFile` is that rustdoc won't have to load the source files from disk again. I plan to do that as a follow up.

Reply:
@eddyb: This looks good to me: were you thinking of imported spans/hygiene?

Reply:
Mostly knowing the `CrateNum` a file came from.